pmd-java/src/main/java/net/sourceforge/pmd/lang/java/metrics/impl/visitors/CycloBaseVisitor.java

```diff
@@ -25,12 +25,12 @@
  * @author Clément Fournier
  * @see net.sourceforge.pmd.lang.java.metrics.impl.CycloMetric
  */
-public class CycloBaseVisitor extends JavaParserControllessVisitorAdapter {
+public final class CycloBaseVisitor extends JavaParserControllessVisitorAdapter {
```

Comment:
Visitors from this package were originally intended to be extended by users to make their custom metrics. It's even [documented on the website](https://pmd.github.io/pmd-6.1.0/pmd_userdocs_extending_metrics_howto.html#best-practices). Any reasons why we may want to change that?

Reply:
Good catch! The rule that triggered here as [ClassWithOnlyPrivateConstructorsShouldBeFinal](https://pmd.github.io/pmd-6.1.0/pmd_rules_java_design.html#classwithonlyprivateconstructorsshouldbefinal).
Then I'll make the default constructors of these visitor classes protected (otherwise you can't subclass them, if there is no accessible constructor....)

Reply:
Or I'll remove the (empty) constructor alltogether - similar to what we have in [AtfdBaseVisitor](https://github.com/pmd/pmd/blob/master/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/metrics/impl/visitors/AtfdBaseVisitor.java) and we get the default constructor back.